usl_pipeline/cloud_functions/main.py

```diff
@@ -563,6 +564,16 @@ def _process_wps_feature(feature: xarray.DataArray, var_config: dict) -> NDArray
     if "num_metgrid_levels" in feature.dims:
         feature = feature.isel(num_metgrid_levels=0)
 
+    # Monthly climatology var - extract to only the month of the file's datestamp
+    if "z-dimension0012" in feature.dims and snapshot_time:
+        snapshot_datetime = datetime.datetime.strptime(
+            snapshot_time.values[0].decode(), "%Y-%m-%d_%H:%M:%S"
+        )
+        feature = feature.isel({"z-dimension0012": snapshot_datetime.month - 1})
+
+    # Drop time axis, no longer needed
```

Comment:
Was it ever needed? Does the code still work if `feature = feature.isel(Time=0)` is up in its original location since we're getting the time from `feature.coords` ?

Reply:
I moved this towards the bottom once we're finished with transforms that need reference to Time (for exaple, this monthly map extraction). If we keep it at the top, we'll remove the Time dim and the corresponding coords metadata

Reply:
1. Attach the wps snapshot as a coordinate (so that we can reference it as metadata attached at the DataArray level) - without this step, snapshot time is only availlable at dataset-level
```py
ds = ds.assign_coords(Time=ds.Times)
```

2. Get the time and do something with it during feature engineering if the feature eng step requires ref to snpashot time (for ex: processing z-dim)
```py
snapshot_time = feature.coords.get("Time")
```

3. At some point at the end of feature eng processing, we can drop the time axis (shape=1) since we don't need that extra dimension in the numpy array for ml modeling. this is performed on all feature matrices
```py
feature = feature.isel(Time=0)
```
^ currently, I'm using this because it was easy to use this xarray method to remove a specifc axis using a standard name instead of just an index. But could choose to do this all the way at the end (line 625) if we wanted to. LMK if that helps

Reply:
The thing I'm specifically confused about is when you say

> we'll remove the Time dim and the corresponding coords metadata

that makes it sounds like `feature = feature.isel(Time=0)` removes the time from the coordinates. Is that true? Can you still access `feature.coords.get("Time")` after doing `feature = feature.isel(Time=0)` ?

Reply:
> that makes it sounds like feature = feature.isel(Time=0) removes the time from the coordinates. Is that true?
Yeah it removes time from the feature dataarray

> Can you still access feature.coords.get("Time") after doing feature = feature.isel(Time=0) ?
No so that's why i moved `feature = feature.isel(Time=0)` from the top since we wanted to do some feature processing that still need reference to the time

Reply:
Okay, so would that affect other features, then? If we loop over a second feature, would it still be able to access `feature.coords.get("Time")` if the previous loop removed it? Or is `ds = ds.assign_coords(Time=ds.Times)` effectively giving each feature its own copy of the times?

I think this would be helpful to document with some more comments.

Reply:
Ah I see your concern. No, it will not affect other features. `feature = feature.isel(Time=0)` does not mutate the feature DataArray it operates on, it returns a new DataArray. I'll add a comment!